src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_BORDER = 5;
+    private static final int BORDER = 101;
+    private int randomFigureIndex;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        randomFigureIndex = random.nextInt(FIGURE_BORDER);
+        switch (randomFigureIndex) {
+            case 0 :
+                return new Circle(colorSupplier.getRandomColor(), random.nextInt(BORDER));
```

Comment:
```suggestion
                return new Circle(randomColor, randomSide());
```

same for other figures
Create separate randomSide()